src/test/ui/error-codes/E0617.rs

```diff
@@ -22,7 +22,7 @@ fn main() {
         //~^ ERROR can't pass `u16` to variadic function
         //~| HELP cast the value to `c_uint`
         printf(::std::ptr::null(), printf);
-        //~^ ERROR can't pass `unsafe extern "C" fn(*const i8, ...) {printf}` to variadic function
-        //~| HELP cast the value to `unsafe extern "C" fn(*const i8, ...)`
+        //~^ ERROR can't pass `for<'r> unsafe extern "C" fn(*const i8, std::ffi::VaList<'r>, ...) {printf}` to variadic function
```

Comment:
This is my least favorite part of this PR. I'd really like for the use to be completely unaware of the spoofed `VaList` arg, but these errors occur in `typeck` which is after we lower the `hir::CVarArgs` to the `VaList`.

Reply:
You need it to be aware however, since that’s the low-level API. What exactly is the problem here?

Reply:
I just see some potential confusion arising from users that might get this error message. From the users perspective there is no `VaList` in the function signature, but the error message will contain the `VaList`.

Reply:
@dlrobertson You can change the `Debug` printing for function type so it doesn't show... obviously you only want one of `VaList` and `...`. :-)

Reply:
Good idea!